src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -151,97 +132,61 @@ void emberAfGroupsClusterServerInitCallback(EndpointId endpointId)
-    auto groupId     = commandData.groupId;
-    auto endpointId  = commandPath.mEndpointId;
-
-    GroupDataProvider::GroupMapping mapping;
-    EmberAfStatus status = EMBER_ZCL_STATUS_NOT_FOUND;
+    auto fabricIndex             = commandObj->GetAccessingFabricIndex();
+    auto groupId                 = commandData.groupId;
+    GroupDataProvider * provider = GetGroupDataProvider();
+    GroupDataProvider::GroupInfo info;
+    Groups::Commands::ViewGroupResponse::Type response;
     CHIP_ERROR err       = CHIP_NO_ERROR;
-    size_t nameSize      = 0;
+    EmberAfStatus status = EMBER_ZCL_STATUS_NOT_FOUND;
 
     if (emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST && emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST_REPLY)
     {
         return true;
     }
 
-    if (IsFabricGroupId(groupId))
-    {
-        GroupDataProvider * provider = GetGroupDataProvider();
-        VerifyOrReturnError(nullptr != provider, false);
-
-        auto * groupIt = provider->IterateGroupMappings(fabricIndex, endpointId);
-        VerifyOrReturnError(nullptr != groupIt, false);
+    VerifyOrExit(IsFabricGroupId(groupId), status = EMBER_ZCL_STATUS_INVALID_VALUE);
+    VerifyOrExit(nullptr != provider, status = EMBER_ZCL_STATUS_FAILURE);
+    VerifyOrExit(provider->HasEndpoint(fabricIndex, groupId, commandPath.mEndpointId), status = EMBER_ZCL_STATUS_NOT_FOUND);
 
-        while (groupIt->Next(mapping))
-        {
-            if (mapping.group == groupId)
-            {
-                nameSize = strnlen(mapping.name, GroupDataProvider::GroupMapping::kGroupNameMax);
-                status   = EMBER_ZCL_STATUS_SUCCESS;
-                break;
-            }
-        }
-        groupIt->Release();
-    }
-    else
-    {
-        status = EMBER_ZCL_STATUS_INVALID_VALUE;
-    }
+    err = provider->GetGroupInfo(fabricIndex, groupId, info);
+    VerifyOrExit(CHIP_NO_ERROR == err, status = EMBER_ZCL_STATUS_NOT_FOUND);
 
-    {
-        app::ConcreteCommandPath path = { endpointId, Groups::Id, Commands::ViewGroupResponse::Id };
-        TLV::TLVWriter * writer       = nullptr;
-        SuccessOrExit(err = commandObj->PrepareCommand(path));
-        VerifyOrExit((writer = commandObj->GetCommandDataIBTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
-        SuccessOrExit(err = writer->Put(TLV::ContextTag(0), status));
-        SuccessOrExit(err = writer->Put(TLV::ContextTag(1), groupId));
-        SuccessOrExit(err = writer->PutString(TLV::ContextTag(2), mapping.name, static_cast<uint32_t>(nameSize)));
-        SuccessOrExit(err = commandObj->FinishCommand());
-    }
+    response.groupName = CharSpan(info.name, strnlen(info.name, GroupDataProvider::GroupInfo::kGroupNameMax));
+    status             = EMBER_ZCL_STATUS_SUCCESS;
 exit:
-    if (err != CHIP_NO_ERROR)
+    response.groupId = groupId;
+    response.status  = status;
+    err              = commandObj->AddResponseData(commandPath, response);
+    if (CHIP_NO_ERROR != err)
```

Comment:
This does not look right to me.  If we got here with a failure status, we should be sending the status, not this response....

Reply:
I believe this function follows the spec. According to section 1.3.6.2.1. whatever the status is, it should be included in the response message ViewGroupResponse, which contains a `status` field.

Reply:
Ah, good point.  Thank you!